api/src/main/java/com/spotify/ffwd/Plugin.java

```diff
@@ -0,0 +1,27 @@
+package com.spotify.ffwd;
+
+import com.spotify.ffwd.filter.Filter;
+import com.spotify.ffwd.filter.TrueFilter;
+import java.util.Optional;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
+public abstract class Plugin {
```

Comment:
This doesn't actually seem like it needs to be an abstract class, or even exist at all?

It has no state which couldn't be stored in OutputPlugin.
If we do end up duplicating some state, that's not the end of the world either.